### PR TITLE
fix: Iframe rehaul

### PR DESCRIPTION
**Pull request by harshilp24**

### CODE REVIEW

File: website/docs/reference/widgets/iframe.md [hunk 1]
```diff
@@ -1,100 +1,162 @@
...
----
 
 # Iframe
 
-![Use iframes to embed other web pages within your applications](</img/as_iframe_cover.png>)
 
-:::info
-**Ensure you have your Iframe widgets sandboxed. By default, this is enabled for all Appsmith instances. See [Security](/product/security#sandboxing-iframe-widgets) for more information.
-:::
+This page explains how to use the iFrame widget to embed external content, such as apps, websites, or custom widgets, into your Appsmith application, while also providing you with the ability to communicate with them.
```
Comment (danciaclara) [inline website/docs/reference/widgets/iframe.md:5]: Include when to use the Iframe here

File: website/docs/reference/widgets/iframe.md [hunk 2]
```diff
@@ -1,100 +1,162 @@
...
 
-## Properties
+<VideoEmbed host="youtube" videoId="4jXTyxUhnP0" title="Using the Iframe widget" caption="Using the Iframe widget"/>
 
-Properties allow you to edit the widget, connect it with other widgets and customize the user actions.
 
-### Widget properties
 
-These properties allow you to edit the Iframe widget. All these properties are present in the property pane of the widget. The following table lists all the widget properties.
+## Display data
```
Comment (danciaclara) [inline website/docs/reference/widgets/iframe.md:11]: remove this

File: website/docs/reference/widgets/iframe.md [hunk 3]
```diff
@@ -1,100 +1,162 @@
...
-Let’s take a closer look at the **message** property.
+<figure>
+  <img src="/img/iframe-website.png" style= {{width:"700px", height:"auto"}} alt="Display external website"/>
+  <figcaption align = "center"><i></i></figcaption>
+</figure>
 
-#### Message
 
-The iframe widget listens for messages sent from the page embedded within it. When this page sends data via the Javascript [`postmessage()`](https://developer.mozilla.org/en-US/docs/Web/API/Window/postMessage) function, Appsmith receives the message and exposes its content to the user on the iframe’s `message` property.
+However, it's important to note that if the iframe widget also has a **srcDoc** property with a value, the widget's **URL** property would be ignored.
```
Comment (danciaclara) [inline website/docs/reference/widgets/iframe.md:26]: we have to give an example of external app that is embedded as well

File: website/docs/reference/widgets/iframe.md [hunk 4]
```diff
@@ -1,100 +1,162 @@
...
+|---------------------	|:------------------------:	|--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------	|
+| **isVisible**  | Boolean  | This property indicates whether the widget is visible or not.                                                                                                                                                                                         |
+| **message**    | Any  | Contains a message received from the embedded page via the JS [`postMessage()`](https://developer.mozilla.org/en-US/docs/Web/API/Window/postMessage) method. It may be of _any_ type. The property is `undefined` before a message is received. |
+| **messageMetadata**    | Any  | 	Contains metadata associated with the message received from the embedded page via the JavaScript `postMessage()` method.  |
```
Comment (danciaclara) [inline website/docs/reference/widgets/iframe.md:137]: Check the any datatype. It should be specific.

**Review by danciaclara — CHANGES_REQUESTED** [review]
Add the security callout. Reword it

**Comment by danciaclara** [comment]
We need to add a callout somewhere that says that if the external page does not allow embedding if X-Frame-Options is set to DENY

https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Frame-Options

File: website/docs/reference/widgets/iframe.md [hunk 5]
```diff
@@ -1,100 +1,205 @@
...
+</script>
+```
+
+This code listens for a message event and updates the text content of an HTML element with the ID `target` based on the data received in the event. When the message event is triggered, the `querySelector()` method is used to find the HTML element, and its `textContent` property is set to the data received in the event.
+
+You can send the message using the Input widget, you need to set its  `onTextChanged` property to `"Post message"`. Set the `"Message"` to `{{Input1.text }}`and the `"Target iframe"` to `Iframe1`.
+
+Now, when you add text to the Input widget and trigger the `onTextChanged` event, the message would be sent to the embedded page using the `postMessage()` method, and the text would be displayed in the embedded page's Iframe.
+
+<figure>
```
Comment (danciaclara) [inline website/docs/reference/widgets/iframe.md:99]: remove image

File: website/docs/reference/widgets/iframe.md [hunk 6]
```diff
@@ -1,100 +1,205 @@
...
+
+<script>
+ window.addEventListener('message', (event) => {
+    const tgt = document.querySelector("#target")
+        tgt.textContent = event.data
+    });
+</script>
+```
+
+This code listens for a message event and updates the text content of an HTML element with the ID `target` based on the data received in the event. When the message event is triggered, the `querySelector()` method is used to find the HTML element, and its `textContent` property is set to the data received in the event.
```
Comment (danciaclara) [inline website/docs/reference/widgets/iframe.md:93]: can we use the same wording here as this is too verbose and difficult to read- https://appsmith-docs-git-iframe-rehaul-get-appsmith.vercel.app/reference/appsmith-framework/widget-actions/post-message#from-appsmith-to-embedded-page

File: website/docs/reference/widgets/iframe.md [hunk 7]
```diff
@@ -1,100 +1,205 @@
...
-4. Type a string into the iframe’s input box and click the "Send” button. You should see your [Text widget](text.md) update to contain the string that you sent from the iframe.
+Appsmith allows secure communication between different Window objects, including iframes. There are a number of directions in which your apps may need to communicate. Depending on their parent/embed relationship and the message's target recipient, the configuration is a bit different. 
+Refer to this [document](/reference/appsmith-framework/widget-actions/post-message) for information on embedding your applications externally.
 
-:::info
-When a message is received, Appsmith also executes any code or actions set within the iframe’s [`onMessageReceived` ](iframe.md#events)event. Try it out – set an action within the `onMessageReceived` event, and send another message to watch the results.
-:::
 
-### Events
+### From Appsmith to embedded page
```
Comment (danciaclara) [inline website/docs/reference/widgets/iframe.md:76]: PLease add the image from here - https://appsmith-docs-git-iframe-rehaul-get-appsmith.vercel.app/reference/appsmith-framework/widget-actions/post-message#from-appsmith-to-embedded-page

File: website/docs/reference/widgets/iframe.md [hunk 8]
```diff
@@ -1,100 +1,205 @@
...
 
-You can try out the message property by following the steps below:
+You can create a wide range of custom widgets in Appsmith, such as advance video players, data visualizations, custom calendar, social media widgets, animations, chat-bots, and more. The possibilities are endless.  
 
-<VideoEmbed host="youtube" videoId="kDJ56AMsXrM" title="Follow these steps to test out how to receive messages posted from the sites embedded in your iframe!" caption="Follow these steps to test out how to receive messages posted from the sites embedded in your iframe!"/>
+---
+**Example**: to add a custom time picker widget to your application, you can use the `select` element for each component of the time (hour, minute, AM/PM) and a `button` to submit the selected time. Once you have the HTML and CSS for the time picker widget, you can add the following JavaScript function to perform any necessary actions with the selected time:
```
[the remark this answers is not in the document]
Comment (harshilp24) [reply website/docs/reference/widgets/iframe.md:46]: this will give lintr error

**Review by danciaclara — APPROVED** [review]

**Pull request merged**